templates/environment/cf.yml

```diff
@@ -260,279 +168,11 @@ Resources:
-              "s3:GetBucketPolicy",
-              "s3:GetObjectVersionTorrent",
-              "s3:GetObjectAcl",
-              "s3:GetEncryptionConfiguration",
-              "s3:GetBucketRequestPayment",
-              "s3:GetObjectVersionAcl",
-              "s3:GetObjectTagging",
-              "s3:GetMetricsConfiguration",
-              "s3:HeadBucket",
-              "s3:GetBucketPublicAccessBlock",
-              "s3:GetBucketPolicyStatus",
-              "s3:ListBucketMultipartUploads",
-              "s3:GetBucketWebsite",
-              "s3:ListJobs",
-              "s3:GetBucketVersioning",
-              "s3:GetBucketAcl",
-              "s3:GetBucketNotification",
-              "s3:GetReplicationConfiguration",
-              "s3:ListMultipartUploadParts",
-              "s3:GetObject",
-              "s3:GetObjectTorrent",
-              "s3:GetAccountPublicAccessBlock",
-              "s3:ListAllMyBuckets",
-              "s3:DescribeJob",
-              "s3:GetBucketCORS",
-              "s3:GetAnalyticsConfiguration",
-              "s3:GetObjectVersionForReplication",
-              "s3:GetBucketLocation",
-              "s3:GetObjectVersion",
-              "kms:Decrypt"
-            ]
-            Resource: "*"
-          - Sid: Tags
-            Effect: Allow
-            Action: [
-              "tag:GetResources"
-            ]
-            Resource: "*"
-          - Sid: DeleteRoles
-            Effect: Allow
-            Action: [
-              "iam:DeleteRole",
-              "iam:ListRolePolicies",
-              "iam:DeleteRolePolicy"
-            ]
-            Resource:
-              - !GetAtt CloudformationExecutionRole.Arn
-              - !Sub "arn:aws:iam::${AWS::AccountId}:role/${AWS::StackName}-EnvManagerRole"
-          - Sid: DeleteEnvStack
-            Effect: Allow
-            Action:
-              - 'cloudformation:DescribeStacks'
-              - 'cloudformation:DeleteStack'
-            Resource:
-              - !Sub 'arn:aws:cloudformation:${AWS::Region}:${AWS::AccountId}:stack/${AWS::StackName}/*'
-
-  # DNS Delegation Resources
-  CertificateValidationFunction:
-    Type: AWS::Lambda::Function
-    Condition: DelegateDNS
-    Properties:
-      Code:
-        ZipFile: |
-          {{.ACMValidationLambda}}
-      Handler: "index.certificateRequestHandler"
-      Timeout: 600
-      MemorySize: 512
-      Role: !GetAtt 'CustomResourceRole.Arn'
-      Runtime: nodejs10.x
-
-  DNSDelegationFunction:
-    Type: AWS::Lambda::Function
-    Condition: DelegateDNS
-    Properties:
-      Code:
-        ZipFile: |
-          {{.DNSDelegationLambda}}
-      Handler: "index.domainDelegationHandler"
-      Timeout: 600
-      MemorySize: 512
-      Role: !GetAtt 'CustomResourceRole.Arn'
-      Runtime: nodejs10.x
+{{include "cfn-execution-role" . | indent 2}}
+
+{{include "environment-manager-role" . | indent 2}}
+
+{{include "custom-resources-role" . | indent 2}}
```

Comment:
Omg this is so good 🤩 , thank you so much for splitting them up into multiple files!